File: test/benchmarks/lua/list.lua
local function reverse(a)
  local len = #a
  local max = math.floor(len / 2)
  for i = 1, max do
    local n = len + 1 - i
    local t = a[i]
    a[i] = a[n]
    a[n] = t
  end
end

local start = os.clock()

local N = 20000000
local list = {}

for i = 1, N do
  list[i] = i - 1
end

reverse(list)

local seconds = os.clock() - start
print('elapsed: ' .. seconds .. 's | ' .. list[1])

File: test/benchmarks/node/list.js
const process = require('process')

function reverse(a) {
  const size = a.length
  const max = Math.floor(size / 2)
  for (let i = 0; i < max; i++) {
    const n = size - i - 1
    const t = a[i]
    a[i] = a[n]
    a[n] = t
  }
}

const start = process.hrtime.bigint()

const N = 20000000
const list = []

for (let i = 0; i < N; i++) {
  list.push(i)
}

reverse(list)

const seconds = Number(process.hrtime.bigint() - start) / 1000000000
console.log('elapsed: ' + seconds + 's | ' + list[0])
